### PR TITLE
fix(main): Code quality issues

### PR DESCRIPTION
**Pull request by wizard-28**
## Purpose

Many code quality issues in the main file.

## Approach

Fix them

## Progress

- [x] `SC2181`
- [x] `SC2236`
- [x] `SC2155`
- [x] `SC2164`
- [x] `[]` -> `[[]]` for all `if`s

### CODE REVIEW

File: pacstall [hunk 1]
```diff
@@ -25,14 +25,17 @@
 # Configuration
 export LC_ALL=C
 export LOGDIR="/var/log/pacstall/metadata"
-export LOGFILE="/var/log/pacstall/error_log/$(date +"%F_%T.log")"
+LOGFILE="/var/log/pacstall/error_log/$(date +"%F_%T.log")"
+export LOGFILE
```
Comment (Henryws) [inline pacstall:29]: ????

File: pacstall [hunk 2]
```diff
@@ -25,14 +25,17 @@
...
 export STGDIR="/usr/share/pacstall"
 export STOWDIR="/usr/src/pacstall"
 
 # Colors
-export BOLD=$(tput bold)
-export NORMAL=$(tput sgr0)
+BOLD=$(tput bold)
+export BOLD
+NORMAL=$(tput sgr0)
+export NORMAL
```
Comment (Henryws) [inline pacstall:38]: Why not on the same line?

File: pacstall (hunk 1 above)
Comment (wizard-28) [reply pacstall:29]: Fixes https://github.com/koalaman/shellcheck/wiki/SC2155

File: pacstall (hunk 2 above)
Comment (wizard-28) [reply pacstall:38]: Fixes https://github.com/koalaman/shellcheck/wiki/SC2155

File: pacstall (hunk 1 above)
Comment (wizard-28) [reply pacstall:29]: It's better for future tests, and the CI

File: pacstall (hunk 2 above)
Comment (wizard-28) [reply pacstall:38]: It's better for future tests, and the CI

File: pacstall (hunk 1 above)
Comment (Henryws) [reply pacstall:29]: > Fixes https://github.com/koalaman/shellcheck/wiki/SC2155

Got it

**Pull request merged**